src/orange/BoundingBox.hh

```diff
@@ -22,60 +22,73 @@ namespace celeritas
 /*!
  * Axis-aligned bounding box.
  *
- * This is currently a pretty boring class; it will be extended as needed.
- * We'll add a BoundingBoxUtils class for all the fancy operations that need
- * only the accessors we provide.
- *
  * The bounding box can be constructed in an unassigned state, in which lower
  * and upper cannot be called.
  */
+template<class T>
 class BoundingBox
 {
   public:
+    //!@{
+    //! \name Type aliases
+    using value_type = T;
+    using array_type = Array<value_type, 3>;
```

Comment:
Hmm, do we use `Real3` or allow it to be more generic? Looking at the class again, there are a couple things that won't compile if we use integers (the nan and infinity accessors). So maybe we should just assume/require it's for real numbers for the time being (sorry to flip-flop on this... I didn't recall the full picture last time).

Even if we leave this as generic, since `array_type` isn't a standard type definition like `value_type` is, maybewe should write as `Array3`?

Reply:
I'll change it to Array3